templates/e3sm_diags.bash

```diff
@@ -82,10 +124,52 @@ param.output_format = {{ output_format }}
 param.output_format_subplot = {{ output_format_subplot }}
 param.multiprocessing = {{ multiprocessing }}
 param.num_workers = {{ num_workers }}
+params = [param]
+
+{%- if "enso_diags" in sets %}
+enso_param = EnsoDiagsParameter()
+enso_param.reference_data_path = '{{ obs_ts }}'
+enso_param.test_data_path = test_ts
+enso_param.test_name = short_name
+enso_param.test_start_yr = start_yr
+enso_param.test_end_yr = end_yr
+enso_param.ref_start_yr = ref_start_yr
+enso_param.ref_end_yr = ref_start_yr + 10
+params.append(enso_param)
+{%- endif %}
+
+
+{%- if "qbo" in sets %}
+qbo_param = QboParameter()
+qbo_param.reference_data_path = '{{ obs_ts }}'
+qbo_param.test_data_path = test_ts
+qbo_param.test_name = short_name
+qbo_param.test_start_yr = start_yr
+qbo_param.test_end_yr = end_yr
+qbo_param.ref_start_yr = ref_start_yr
+qbo_param.ref_end_yr = ref_start_yr + num_years - 1
+if (qbo_param.ref_end_yr <= {{ ref_final_yr }}):
+  params.append(qbo_param)
+{%- endif %}
+
+
+{%- if "area_mean_time_series" in sets %}
+ts_param = AreaMeanTimeSeriesParameter()
+ts_param.reference_data_path = '{{ obs_ts }}'
+ts_param.test_data_path = test_ts
+ts_param.test_name = short_name
+ts_param.start_yr = start_yr
+ts_param.end_yr = end_yr
+params.append(ts_param)
+{%- endif %}
 
 # Run
-runner.sets_to_run = {{ sets }}
-runner.run_diags([param])
+runner.sets_to_run = [{{ sets }}]
+{%- if "qbo" in sets %}
+if (qbo_param.ref_end_yr > {{ ref_final_yr }}):
+  runner.sets_to_run.remove('qbo')
```

Comment:
On second thought, I think https://github.com/E3SM-Project/e3sm_diags/issues/412 might not actually be an issue at all. I think in this conditional we just need to set `ref_end_yr` to `ref_final_yr`, not remove QBO diags completely. Testing now on `e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_0001-0050.bash`

Reply:
No, it looks like even with setting the end year explicitly, we still get `TypeError: Shapes of x (72, 456) and z (72, 600) do not match`. We'll have to fix that on the E3SM Diags side.

Reply:
@golaz https://github.com/E3SM-Project/e3sm_diags/pull/414 merged, meaning that we should be able to run QBO Diags even on the 50 year sets (because there is only 38 years of ref data, QBO diags will actually only cover 38 years). However, that change is not yet in the unified environment. That said, that change should be included with the first unified environment `zppy` is a part of. So, should we just remove these two lines now?